### ISSUE TITLE
Routing with Two Different Sites

### ISSUE THREAD

**Issue by aatmmr** [PiranhaCMS/piranha.core#437]
Hej Community!

I got somewhat lost and can't wrap my head around the following:

I have a simple scenario with 2 Sites, the `DefaultSite` and a custom `MaintenanceSite`. Each has a single page with a couple of Fields, such as, `StringField` etc. Her comes the problem: I cannot access the page of the Site that is not set as default. If I call the slug of the non-default page / site directly the respective Action is called, however, an `id` of `00000000-0000-0000-0000-000000000000` is passed to the action.

The pages are both available in the database with the expected slug.

In `CmsController.cs` I have the following Actions to get the request
```csharp
[Route("maintenance")]
public IActionResult MaintenancePage(Guid id) 
{
     var model = _api.Pages.GetById<Web.Models.Pages.MaintenancePage>(id);
     return View(model);
}

[Route("test")]
public IActionResult TestPage(Guid id)
{
     var model = _api.Pages.GetById<Web.Models.Pages.TestPage>(id);
     return View(model);
}
```

The version of Piranha is the latest, namely `5.2.1`.

Am I misunderstanding anything?! I checked the routing documentation and as I far as I understood, the slug should be found and the route generated accordingly. 

Any help appreciated :)

**Comment by tidyui** [comment]
Hi there! You can't have more than one site on the same hostname. If the pipeline can't any site that specifically specified the requested hostname then the default site is used (this is the standard case). You can use sites for several purposes, for example subdomains:

* Default Site (empty hostnames)
* Maintenance Site (maintenance.myhostname.com)

Or folder/suffix mapping for language sites, for example

* Default Site (empty hostnames)
* Swedish Site (myhostname.com/se)

**Comment by aatmmr** [comment]
Thank you for the quick reply! I tried both, however, no success yet.

I set the hostnames / sites as following (specific dev example):

   - Default Site (empty hostnames) - set also as default of installation
   - Maintenance Site (localhost:5001/maintenance)

When I reload the page at `localhost:5001` the maintenance page is loaded. If I specifically call `localhost:5001/test` the controller action is found but again with `id` `00000000-0000-0000-0000-000000000000`.

The ultimate goal is actually subsites for languages and my thinking was to catch the user culture / language and route to the matching subsite.
